### PR TITLE
Refactor `inquirer` package

### PR DESCRIPTION
**Pull request by PopGoesTheWza**
Fixes no specific issue

- [x] `npm run test` succeeds.
- [x] `npm run lint` succeeds.
- [ ] Appropriate changes to README are included in PR.

### CODE REVIEW

**Comment by PopGoesTheWza** [comment]
@grant @takanakahiko I did not do extensive tests and I am unsure if builtin tests coverage is enough.

File: src/inquirer.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    type: 'list',
+  },
+]);
+
+/**
+ * Inquirer prompt for script type.
+ * @returns {Promise<{ type: string }>} A promise for an object with the `type` property.
+ */
+export const scriptTypePrompt = () => prompt<{ type: string }>([{
+  choices: Object.keys(SCRIPT_TYPES).map((key) => SCRIPT_TYPES[key as any]), // tslint:disable-line: no-any
```
Comment (grant) [inline src/inquirer.ts]: Try: `(key: string)`

No `key as any`. (no-any)
Maybe `key as string` if required.

Comment (PopGoesTheWza) [reply src/inquirer.ts]: The correct cast to access elements of an enum is the ugly `[key as unknown as number]`

**Comment by PopGoesTheWza** [comment]
@grant merge now, if you wish

**Comment by PopGoesTheWza** [comment]
@grant: I spoted several instances of unnecessary `exports` as well as a couple unused functions. Should I commit with the extra exports removed and the unused ones flagged with the comment `// TODO: currently unused. Check relevancy` ?

**Pull request merged**